### ISSUE TITLE
zebrad: command-line help info is hard to access and understand 

### ISSUE THREAD

**Issue by Rqnsom** [ZcashFoundation/zebra#5624]
## Description

Initially, I wanted to check the description of the `--verbose` option since one of our scripts in the Ziggurat project uses that option.

I had to find the zebrad's help output, but to find out what the help subcommand looks like, I just used an invalid subcommand:

```sh
➜  codecs git:(main) ✗ zebrad random-things
error: unrecognized command `random-things`

zebrad 1.0.0-rc.0
Zcash Foundation <zebra@zfnd.org>

FLAGS:
    -c, --config CONFIG       path to configuration file
    -h, --help                print help message
    -v, --verbose             be verbose
```

After seeing `    -h, --help                print help message`,  I used `--help` with no effect. Not only did I not get the help text, it even started the process.
```sh
➜  codecs git:(main) ✗ zebrad --help
2022-11-13T12:20:45.661166Z  INFO zebrad::components::tracing::component: started tracing component filter="info" TRACING_STATIC_MAX_LEVEL=LevelFilter::INFO LOG_STATIC_MAX_LEVEL=Info
2022-11-13T12:20:45.661203Z  INFO zebrad::application: Diagnostic Metadata:
version: 1.0.0-rc.0
Zcash network: Mainnet
state version: 25
branch: master
git commit: a4cb835
commit timestamp: 2022-10-12T01:17:55Z
target triple: aarch64-apple-darwin
build profile: release
[...]
```
Then I realized the command `zebrad help` would give me what I needed. And there I also saw that the `verbose` flag doesn't exist anymore.

### Steps to Reproduce

Run:
```sh
zebrad invalid-input
```
or
```sh
zebrad --helpppp
```
but not (could be considered as a separate issue):
```sh
zebrad --help # this one actually starts the zebrad process - it's probably the same root cause that causes this behaviour
```

### Expected Behaviour

I expected that the invalid subcommand input would result in the `zebrad help` help text output.

### Actual Behaviour

Instead, this happened:
I got the obsolete help text with the wrong information:
```
zebrad 1.0.0-rc.0
Zcash Foundation <zebra@zfnd.org>

FLAGS:
    -c, --config CONFIG       path to configuration file
    -h, --help                print help message
    -v, --verbose             be verbose
```

### Zebra Logs

N/A

### Zebra Version
```sh
➜  ~ zebrad version
zebrad 1.0.0-rc.0
```
### Operating System
```sh
➜  ~ uname -a
Darwin EQML220902-02 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:20:07 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T8110 arm64
```

## Designs

- Print the same information for invalid commands or options as is printed by the `help` command/flag

**Comment by arya2** [comment]
The `--verbose` flag is still being used to set the tracing filter to `debug`. It seems top-level options are omitted by the `help` command, and the only way to see them is by trying an invalid command or option.

The part where the `--help` flag on its own is running the start command is related to https://github.com/ZcashFoundation/zebra/pull/4957 where we changed Zebra to start by default.

On this note, `help <command>` will print the options for that command, but the only way to find that out via the CLI is by using it with `zebrad help help`.

> I expected that the invalid subcommand input would result in the zebrad help help text output.

I agree with this expected behaviour. It seems unrecognized options are being handled by gumdrop as parse errors, so we may want to leave that part until migrating to abscissa 0.6.0 and clap (issue https://github.com/ZcashFoundation/zebra/issues/5502).

**Comment by Rqnsom** [comment]
> The --verbose flag is still being used to set the tracing filter to debug.

Okay, thanks for the info and for the clarification!

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/5624?planning-poker) with Zenhub @arya2 @conradoplg @dconnolly @oxarbitrage @teor2345

**Comment by teor2345** [comment]
I'm not sure what still needs to be done for this ticket, it looks like it got partly fixed in PR #5634.

@arya2 can you add a list of the fixes that are still needed at the top of the ticket?

**Comment by arya2** [comment]
I updated this with a vague design and split out the part of this that's easy to address without updating Abscissa out to issue #5707
